### DISCUSSION TITLE
Query Asynchronously

### DISCUSSION THREAD

**Discussion by vbrm** [prisma/prisma#23682] [Q&A] [answered]
### Question

Does (generated) prisma client support querying asynchronously?  

Use case: 
Prisma querying feature sits behind REST API (e.g., AWS API gateway ) where timeout is 30 seconds.   
Is there a feature capability where prisma query engine returns a reference `id` so that, client could come back for either status or results?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @vbrm 👋 

Thank you for raising this question.

> Does (generated) Prisma client support querying asynchronously?

Yes, Prisma Client fully supports asynchronous querying through Promises. 

> Is there a feature capability where Prisma query engine returns a reference id so that, client could come back for either status or results?

Prisma Client doesn't natively return reference IDs for ongoing queries. It doesn't provide built-in support for managing asynchronous queries with reference IDs out of the box, as this functionality is typically handled at the application level. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
